### PR TITLE
UCP/GTEST: added bitwise atomic tests

### PR DESCRIPTION
**Pull request by hoopoepg**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6879/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6881/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6884/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
failed due to timeout

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6892/ for details (Mellanox internal link).

File: test/gtest/ucp/test_ucp_atomic.cc [hunk 1]
```diff
@@ -165,24 +165,45 @@ void test_ucp_atomic::blocking_cswap(entity *e,  size_t max_size, void *memheap_
...
 }
 
+template <typename T, ucp_atomic_post_op_t OP>
+void test_ucp_atomic::nb_post(entity *e,  size_t max_size, void *memheap_addr,
+                              ucp_rkey_h rkey, std::string& expected_data)
+{
+    ucs_status_t status;
+    T val, prev;
+
+    prev   = *(T*)memheap_addr;
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_atomic.cc]: isn't this copy-paste from nb_add? what is the difference?

File: test/gtest/ucp/test_ucp_atomic.h [hunk 2]
```diff
@@ -59,11 +59,57 @@ class test_ucp_atomic : public test_ucp_memheap {
...
+            return 0;
+        }
+    }
+
+    template <typename T, ucp_atomic_fetch_op_t OP>
+    T atomic_fop_val(T v1, T v2)
+    {
+        /* coverity[switch_selector_expr_is_constant] */
+        switch (OP) {
+        case UCP_ATOMIC_FETCH_OP_FADD:
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_atomic.h]: i already saw this is another test (fence i think), can it go to some common code?

File: test/gtest/ucp/test_ucp_atomic.cc [hunk 3]
```diff
@@ -218,6 +239,29 @@ ucs_status_ptr_t test_ucp_atomic::ucp_atomic_fetch(ucp_ep_h ep,
...
+                               void *memheap_addr, ucp_rkey_h rkey,
+                               std::string& expected_data)
+{
+    void *amo_req;
+    T val, prev, result;
+
+    prev    = *(T*)memheap_addr;
+    val     = (T)ucs::rand() * (T)ucs::rand();
+
+    amo_req = test_ucp_atomic::ucp_atomic_fetch<T>(e->ep(), FOP,
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_atomic.cc]: same here, need to unite code with nb_add

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6894/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
it seems there are issues on arm

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6895/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6896/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6903/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
@yosefe anything else to fix?

**Pull request merged**